Tag environmental sustainability (#903)
cncf/toc: tags/environmentalsustainability.md

# CNCF TAG Environmental Sustainability

## Introduction

The Paris Climate Accords outline the link between economic activity, greenhouse gas (GHG) emissions, and the impacts of climate change. GHG  emissions, primarily made from Carbon Dioxide, are released during the combustion of fossil fuels to produce electricity. To reach net zero goals, we will first identify how to reduce the environmental footprints of increasingly large data centers and then help the community take action to mitigate these footprints.

References:

* https://www.iea.org/reports/net-zero-by-2050
* https://www.whitehouse.gov/wp-content/uploads/2021/10/US-Long-Term-Strategy.pdf
* https://www.worldbank.org/en/news/feature/2022/05/23/what-you-need-to-know-about-net-zero

## Mission Statement

> This TAG's goal is to advocate for, develop, support, and help evaluate environmental sustainability initiatives in Cloud Native Technologies.  This TAG will identify values and possible incentives for service providers to reduce their consumption and carbon footprint through cloud native tooling.

Reviewer: ```suggestion
This TAG's goal is to advocate for, develop, support, and help evaluate environmental sustainability initiatives in cloud native technologies.  This TAG will identify values and possible incentives for service providers to reduce their consumption and carbon footprint through cloud native tooling.
```

## Responsibilities & Deliverables

### Background
We recognize that: 
- the trade off between reducing resource consumption and higher performance is hard to balance

> - there is a lack of well supported and documented approaches for evaluating cloud native projects in their default or optimized configurations

Reviewer 2: ```suggestion
- there is a lack of well supported and documented approaches for evaluating the efficiency of cloud native projects in their default or optimized configurations
```

... just because we're not talking, say, fitness for purpose here.

Author: @jberkus  this changes my interpretation of the sentence.  For instance, we may not care about the efficiency in some cases, but may care more about workload positioning, in a cross-cloud instance (transfer between sunny regions with solar power around the world, for instance).  Are you okay with leaving the wording as is, with that consideration?  Or do you have another suggestion?

Reviewer 3: @jberkus Alternative wording proposal that I think may address the intent behind proposing to add "efficiency" but without being overly limiting as per @catblade's point. 
```suggestion
- there is a lack of well supported and documented approaches for evaluating the environmental sustainability of cloud native projects in their default or optimized configurations

```

### In-Scope

- Identify, define, and develop tooling to assess and improve environmental sustainability approaches, including

>     - Quantify the energy consumption of cloud native implementations individually as well as in common integration patterns.

Reviewer: ```suggestion
    - Quantify the energy consumption of cloud native implementations individually as well as in common integration patterns
```
```suggestion
    - Quantify the energy consumption of cloud native implementations individually as well as in common integration patterns
```

    - Recommendations and strategies to develop, package, distribute, deploy, and operate cloud native implementations to reduce energy consumption and carbon as well as the various environments they operate in and which impact their consumption
    - Capabilities, benchmarks, and processes to evaluate technological and architectural health of projects

> - Community outreach and engagement on the work of this TAG.

Reviewer: ```suggestion
- Community outreach and engagement on the work of this TAG
```

> - Collaboration with other environmental or sustainability organizations, initiatives, activities, and efforts that may fall outside of the Foundation

Reviewer: ```suggestion
- Collaboration with other environmental or sustainability organizations, initiatives, activities, and efforts that may fall outside of the CNCF (Cloud Native Computing Foundation)
```

### Out of scope
- Form an umbrella organization beyond the CNCF
- Establish a compliance and standards body beyond the CNCF space
- Evaluate individual company infrastructures

> - Focus outside of Cloud Native Technologies, according to the [CNCF Cloud Native definition](https://github.com/cncf/toc/blob/main/DEFINITION.md)

Reviewer: ```suggestion
- Focus outside of cloud native technologies, according to the [CNCF Cloud Native definition](https://github.com/cncf/toc/blob/main/DEFINITION.md)
```

> ## Deliverables to ToC

Reviewer: ```suggestion
## Deliverables to TOC
```

- Landscape for carbon and energy efficiency in the form of metrics, measurements, and management techniques
- Environmental sustainability recommendations and optimizations to new and existing projects within the landscape
- Reports on gaps in the environmental sustainability coverage of the landscape
- Reviews, inputs, and recommendations for proposed projects for CNCF hosting and advancement
- Suggestions for improvements for CNCF internal processes, for example, education for sustainability

## Audiences

- Education - audience is end users, developers, stakeholders
- Project intelligence - audience is TOC/CNCF Community

> - External collaboration - organizations, initiatives, activities, and efforts outside of the foundation

Reviewer: ```suggestion
- External collaboration - organizations, initiatives, activities, and efforts outside of CNCF (Cloud Native Computing Foundation)
```

### Out of scope
- Form an umbrella organization beyond the CNCF
- Establish a compliance and standards body beyond the CNCF space
- Evaluate individual company infrastructures
- Focus outside of Cloud Native Technologies, according to the [CNCF Cloud Native definition](https://github.com/cncf/toc/blob/main/DEFINITION.md)

## Deliverables to ToC
- Landscape for carbon and energy efficiency in the form of metrics, measurements, and management techniques
- Environmental sustainability recommendations and optimizations to new and existing projects within the landscape
- Reports on gaps in the environmental sustainability coverage of the landscape
- Reviews, inputs, and recommendations for proposed projects for CNCF hosting and advancement
- Suggestions for improvements for CNCF internal processes, for example, education for sustainability

## Audiences

- Education - audience is end users, developers, stakeholders
- Project intelligence - audience is TOC/CNCF Community
- External collaboration - organizations, initiatives, activities, and efforts outside of the foundation

> #### Operations

Reviewer 4: ```suggestion
## Operations
```

Environmental Sustainability TAG operations are consistent with standard TAG operating guidelines provided by the [CNCF Technical Oversight Committee TOC](https://github.com/cncf/toc).

> 
> 
> 

Reviewer 5: ```suggestion
```

# CNCF TAG Environmental Sustainability

## Introduction

> The Paris Climate Accords outline the link between economic activity, greenhouse gas (GHG) emissions, and the impacts of climate change. GHG  emissions, primarily made from Carbon Dioxide, are released during the combustion of fossil fuels to produce electricity. To reach net zero goals, we need to identify how to reduce the environmental footprints of increasingly large data centers.

Reviewer 6: Isn't reaching net zero GHG emissions just one goal?

Reviewer 7: The term `net zero goals` is frequent used [1,2,3] to represent a set of owners, contexts, and objectives. 

1. https://www.iea.org/reports/net-zero-by-2050
2. https://www.whitehouse.gov/wp-content/uploads/2021/10/US-Long-Term-Strategy.pdf
3. https://www.worldbank.org/en/news/feature/2022/05/23/what-you-need-to-know-about-net-zero

Reviewer 6: Alright then.

Reviewer 6: Thank you very much for the clarification.

> The Paris Climate Accords outline the link between economic activity, greenhouse gas (GHG) emissions, and the impacts of climate change. GHG  emissions, primarily made from Carbon Dioxide, are released during the combustion of fossil fuels to produce electricity. To reach net zero goals, we need to identify how to reduce the environmental footprints of increasingly large data centers.

Reviewer 4: `To reach net zero goals, we need to identify how to reduce the environmental footprints of increasingly large data centers.`

- As written this implies "identify" is enough to reach "net zero goals"? Isn't it just one step?
- "We" - tag? users of cloud native technologies? everyone who has a data center? 
- Do we want to "mitigate" as well? in addition to "identify"?

Reviewer 6: I think you have your answer right there, "we" must refer to the TAG, but since it is a global problem, I think it is actually correct to say "we" instead of the "the TAG".

Author: @dims  I updated here:  https://github.com/catblade/toc/pull/3

I have not pushed yet, because I'm waiting for the community to respond.

How about the wording:
we need to first identify how to reduce the environmental footprints of increasingly large data centers and then help the community take action to mitigate these footprints.

Reviewer 4: @catblade sounds good!

Reviewer 8: Since this is a charter of the TAG, suggest to change to "The TAG will first ..." unless "we" can be clearly defined and people involved in the "we" are committed to the work.

Reviewer 8: Better to define the scope of actions this TAG will cover to mitigate these footprints.

Author: @cathyhongzhang  we have a scope section-it has many items defined there.

Author: @cathyhongzhang I'm confused on your other comment regarding the "we".  I would assume "we" is the TAG, as it is the TAG charter.  I don't see another real interpretation here.  Am I misunderstanding your comment?

## Mission Statement
This TAG's goal is to advocate for, develop, support, and help evaluate environmental sustainability initiatives in Cloud Native Technologies.

> All activities of this TAG should aim to positively impact the community by changing mindset, increasing resource efficiency, and putting energy efficiency and environmental sustainability on everyone's agenda.

Reviewer 4: "changing mindset" of who? what?

Author: Update, same PR as above, with this suggestion:
All activities of this TAG should aim to positively impact the community by working to help reach net-zero goals and simplifying the processes to do so.

Reviewer 4: way better. thanks

Reviewer 9: > would recommend adding something to the effect of "and strategizing areas of incentive for service providers to reduce their consumption and carbon footprint through cloud native tooling."

Could you explain what you mean here? I am not clear on "Strategizing areas of incentive". Are you suggesting a standard?

Reviewer 5: definitely not standards.  often ive found some groups dont see why they should do something. laying that out for them in an easy way where they understand the value and are subsequently incentivized to apply the change is real valuable in this space, particularly where there is cost savings long term or immediate after initial investment of the change

Reviewer 9: Ok, so maybe revise to: " "and identifying value(s) and possible incentives for service providers to reduce their consumption and carbon footprint through cloud native tooling."

Reviewer 5: Excellent!

Author: Updated.  many thanks.

## Responsibilities & Deliverables

### Background
We recognize that: 

> - balancing resource consumption with the desire for performance is difficult

Reviewer 4: `Trade off between reducing resource consumption and higher performance is hard` << is this better?

Reviewer 6: Could you detail what you prefer in your sentence over the original?

Author: Easier to follow I think.  as a compromise:
the trade off between reducing resource consumption and higher performance is hard to balance

(same PR as above)

Reviewer 4: first part is an apple ("resource consumption") the second part is an orange ("desire for performance")

Author: Problem is, people want apples and oranges on the same box.

Reviewer 4: @catblade "lowering resource consumption" and "higher performance" is comparable ... the problem is the "desire for"

Author: Fair enough.  Will update in the other PR.  Thanks for clarifying.

> - there is inconsistency in how emissions are calculated and what is included by software in scopes of emission. 

Reviewer 4: this sentence is not clear.

Reviewer 6: What seems unclear to you?

Reviewer 4: what does "software in scopes of emission" mean?

Reviewer 7: the `scopes` refer to Scopes 1,2,3 emissions. EPA has [guidelines](https://www.epa.gov/climateleadership/scope-1-and-scope-2-inventory-guidance) on the emission factors

Author: Same PR, suggestion for rewording:
- there is wide inconsistency in how emissions are calculated among projects

Reviewer 8: Agree with Dims. The original sentence is not clear. The reworking one is much better

Reviewer 9: Like that reword @TheFoxAtWork. I will incude in @catblade PR.

### In-Scope

- Identify, define, and develop tooling to assess and improve environmental sustainability approaches, including

>     - quantification of energy consumption of full-stack software including application, cloud native services

Reviewer 9: Will add to the other PR.

>     - quantification of energy consumption of full-stack software including application, cloud native services

Author: Changed and resolved.

>     - recommendations to manage full-stack software such that the overall energy consumption and associated carbon are reduced.

Reviewer 5: "Recommendations and strategies to develop, package, distribute, deploy, and operate cloud native projects that reduce energy consumption and carbon as well as the various environments they operate in and which impact their consumption"

I recommend avoiding full stack here as it can limit your scope to purely operational deployments and not apply to the development.  Enumerating the various phases of software development and software operation here allows you to consider their impact on the next stage. Potentially identifying optimizations at each step that build on each other. Further, consumption on edge compute versus a laptop versus an EC2 instance all vary so environment of deployment and throughout make a huge difference."

>     - items to evaluate technological and architectural health of projects

Reviewer 8: What is meant by "items to evaluate technological and architectural health of projects"? Do "items" refer to tooling or something else? What is meant by "technological and architectural health"?

Author: tooling and/or processes-it's a generic term.  "technological and architectural health" has to do with general soundness of the projects in terms of what is needed.  An example would be the applying something like this: https://github.com/Green-Software-Foundation/software_carbon_intensity to a project, or known measurement models, et cetera.  This sort of science can be hard.

Reviewer 5: I would change "items" to "capabilities, benchmarks, and processes" to add the nuance you are seeking.

As to health, this often has specific meaning in a variety of contexts.  We think of community health of a project, or security health of a project. I'm not opposed to environmental health being introduced in this context however when used be sure to caveat as environmental to avoid confusion.

Reviewer 9: Will add to the PR.

Author: Updated PR with new wording from meeting today.  Many thanks.

> - Community outreach on any items within scope, for instance, social media/conferences/podcasts/meetups et cetera

Reviewer 5: A much simpler rewrite:  "Community outreach and engagement on the work of this TAG."
This means you are not limited to defined items but all work as described in the charter and the operation of the group.

> - Collaboration with other sustainability efforts

Reviewer 4: you mean `with other organizations`?

Reviewer 6: We chose "efforts" here, because we must not artificially limit our self with whom or what we can collaborate.

Author: @dims  it could be organizations, or it could be projects/universities/??? in other areas.  we chose to be general intentionally.  Is there something else that would be more clear on this?

Reviewer 4: got it. felt incomplete that's all.

Author: Any suggestions on what we could add for clarity?

Reviewer 4: @catblade some examples may help

> - Collaboration with other sustainability efforts

Reviewer 5: "Collaboration with other environmental or sustainability organizations, initiatives, activities, and efforts that may fall outside of the Foundation."

Author: (added)

### Out of scope
- Form an umbrella organization beyond the CNCF
- Establish a compliance and standards body beyond the CNCF space
- Evaluate individual company infrastructures
- Focus outside of Cloud Native Technologies, according to the [CNCF Cloud Native definition](https://github.com/cncf/toc/blob/main/DEFINITION.md)

## Deliverables to ToC

> - landscape for the carbon and energy metrics/measurements/management savings

Reviewer 4: what is a "management savings"?

Author: I think we got slash happy.  Looking at this maybe?
landscape for the carbon and energy metrics, measurements, and management

Reviewer 8: Do you mean measurement tools/mechanisms or the real measured data?

Author: Both, hence the "carbon and energy metrics, measurements, and management."  Do you have something you would like added here @cathyhongzhang ?

Reviewer 5: "Landscape for carbon and energy efficiency in the form of metrics, measurements, and management techniques"

Reviewer 9: Will add to the PR.

> - recommendations for new CNCF projects where there are clear use cases or where exploration is needed

Reviewer 4: how about existing?

Author: How about we add this:
- review of upcoming project proposed for adoption and propose to CNCF according to interest

Reviewer 8: How about "review the upcoming project applied for adoption and provide feedback to the CNCF TOC"

Author: Maybe "review the upcoming project applied for adoption and provide feedback to the CNCF TOC according to community interest" ?  I'd lie to know what we are evaluating on.

Reviewer 5: "Provide environmental sustainment recommendations and optimizations to new and existing projects within the landscape and areas of gap in the environmental sustainment coverage of the landscape"

Reviewer 9: will add to the PR to discuss.

Author: Updated PR with new wording from meeting today.

> - regular reporting to ToC on ongoing and completed work

Reviewer 5: TOC is all caps as the O is Oversight.

Reviewer 9: Including on the other PR.

> - recommend projects for sandbox and graduation status

Reviewer 5: "Recommend projects for inclusion into the CNCF".  This includes all phases such that u don't need to rewrite it if we ever change them.

Reviewer 9: +1. Will Add to the other PR.

Author: Added some wording, but should encapsulate that.  There were changes suggested (further) in the meeting today, but should reflect this.

- suggestions for improvements for CNCF internal processes, for example, education for sustainability

## Audiences

- Education - audience is end users, developers, stakeholders

> - Project intelligence - audience is TOC / TOC Community

Reviewer 5: I'm not sure I understand this?  The TOC is not it's own community (although we have a mailing list and repo). Do you mean the CNCF and it's project communities to include the TAGs?

Reviewer 9: Good catch.Yes, CNCF community. Will add to the other PR.

Reviewer 4: please mark this as resolved

Author: @dims  I only see an "unresolve conversation" button at this point?

Reviewer 4: whoops, snafu on my end. no worries.

> - External collaboration - organizations with the same mindset

Reviewer 5: Don't limit this to the same mindset, if people already agreed this is necessary and were doing it we wouldn't be where we are.  You will need to collaborate beyond the echo chamber.

Author: Ate the mindset for lunch.  Updated.

#### Operations

Environmental Sustainability TAG operations are consistent with standard TAG operating guidelines provided by the [CNCF Technical Oversight Committee TOC](https://github.com/cncf/toc).

## Acknowledgement
The format of this proposal is inspired from [TAG security](https://github.com/cncf/tag-security/blob/main/governance/charter.md#operations).
The initial doc was taken from the [Environmental WG charter](https://docs.google.com/document/d/1JaF7lSUmLQ2zmScmca6UF7PgbjMzSxjhhjx2LThThaY/edit#heading=h.x77xxicolnig) and modified by the community.

> ## Contributors

Reviewer 4: please move this to the git commit message. It does not belong here.

Reviewer 5: Agreed with @dims

Reviewer 9: Adding to the other PR.

Author: Removed and added in commits.  Thank you.
